src/ic-cdk/src/api/management_canister/main/types.rs

```diff
@@ -30,6 +30,16 @@ pub struct CreateCanisterArgument {
     pub settings: Option<CanisterSettings>,
 }
 
+#[derive(
+    CandidType, Serialize, Deserialize, Debug, PartialEq, Eq, PartialOrd, Ord, Hash, Clone, Default,
+)]
+pub(crate) struct CreateCanisterArgumentExtended {
```

Comment:
Why not adding the new option field to the original struct? It should be a backward compatible change.

Reply:
> It should be a backward compatible change.

I don't think so as Rust does not allow default values for struct fields. This means that clients of the library would need to add this field to their code (either assigning `None` or `Some(ic_cdk::api::canister_version())` to the field). This PR saves the clients from that hassle.

Reply:
Or we could bump the major version and let the users to update their code, which is a common practice in Rust. If we add a new struct for each API change, it's also confusing to the users which struct to use.